src/exe.ml

```diff
@@ -138,10 +138,26 @@ let link_exe
       Build.dyn_paths (Build.arr (fun (modules, _) ->
         artifacts modules ~ext:ctx.ext_obj))
   in
+  let arg_spec_for_requires, flags =
+    match requires with
+    | Ok libs ->
+      let open Link_time_code_gen in
+      let l = create ~flags:(CC.flags cctx) libs in
+      let l = findlib_dynload ~name ~mode cctx l in
+      let arg_spec =
+        List.map l.libs_or_module ~f:(function
+          | Libs l -> Lib.L.link_flags l ~mode ~stdlib_dir:ctx.stdlib_dir
```

Comment:
`link_flags` does some sharing of flags, so it doesn't seem right to call it multiple time. It should probably take the `lib_or_module list` type directly.